### PR TITLE
Add durable isolated orchestration item template

### PR DESCRIPTION
**Pull request by jviau**
Adds item template for durable functions orchestration in dotnet isolated. The template is a direct copy of the existing template for in-proc, with changes to make it compatible with isolated (using and type changes).

### CODE REVIEW

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/.template.config/template.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  },
+  "primaryOutputs": [
+    {
+      "path": "DurableFunctionsOrchestrationCSharp.cs"
+    }
+  ],
+  "defaultName": "DurableFunctionsOrchestrationCSharp",
+  "postActions": [
+    {
+      "Description": "Adding Reference to Microsoft.Azure.Functions.Worker.Extensions.DurableTask NuGet package",
```
Comment (cgillum) [inline Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/.template.config/template.json:35]: Should "Reference" be lowercased here?

Comment (cgillum) [reply Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/.template.config/template.json:35]: I also noticed the "Description" is capitalized here but lowercase in some of the other post actions.

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/.template.config/template.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      "ContinueOnError": "true",
+      "ManualInstructions": [],
+      "args": {
+        "referenceType": "package",
+        "reference": "Microsoft.Azure.Functions.Worker.Extensions.DurableTask", "version": "1.0.0-rc.1",
+        "projectFileExtensions": ".csproj"
+      }
+    },
+    {
+      "description": "Restore NuGet packages required by this project.",
```
Comment (cgillum) [inline Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/.template.config/template.json:46]: This description ends in a period, but the others don't. Which is correct?

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        public static async Task<HttpResponseData> HttpStart(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post")] HttpRequestData req,
+            [DurableClient] DurableClientContext durableContext,
+            FunctionContext executionContext)
+        {
+            ILogger logger = executionContext.GetLogger("DurableFunctionsOrchestrationCSharp_HttpStart");
+
+            // Function input comes from the request content.
+            string instanceId = await durableContext.Client
+                .ScheduleNewOrchestrationInstanceAsync("DurableFunctionsOrchestrationCSharp");
```
Comment (cgillum) [inline Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs:42]: I recommend we use `nameof` for referencing the orchestration, just like we do for the activity calls.

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+            logger.LogInformation($"Started orchestration with ID = '{instanceId}'.");
+
+            return durableContext.CreateCheckStatusResponse(req, instanceId);
```
Comment (cgillum) [inline Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs:46]: Maybe add a comment explaining that this returns an HTTP 202 response with links to instance management HTTP APIs?

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+{
+    public static class DurableFunctionsOrchestrationCSharp
+    {
+        [FunctionName("DurableFunctionsOrchestrationCSharp")]
+        public static async Task<List<string>> RunOrchestrator(
+            [OrchestrationTrigger] TaskOrchestrationContext context)
+        {
+            var outputs = new List<string>();
+
+            // Replace "hello" with the name of your Durable Activity Function.
```
Comment (cgillum) [inline Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs:16]: This comment doesn't seem to match the code.

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/function.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "bindings": [
+    {
+      "name": "context",
+      "type": "orchestrationTrigger",
```
Comment (cgillum) [inline Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/function.json:5]: What is this file for?

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/.template.config/template.json (hunk 1 above)
Comment (jviau) [reply Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/.template.config/template.json:35]: Lowercased "reference". Left "Description" as is, as that is consistent with all other `template.json` files and with the other properties in this section.

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/.template.config/template.json (hunk 2 above)
Comment (jviau) [reply Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/.template.config/template.json:46]: Added a period to the other descriptions.

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+{
+    public static class DurableFunctionsOrchestrationCSharp
+    {
+        [FunctionName("DurableFunctionsOrchestrationCSharp")]
+        public static async Task<List<string>> RunOrchestrator(
+            [OrchestrationTrigger] TaskOrchestrationContext context)
+        {
+            var outputs = new List<string>();
+
+            // Replace name and input with values relevant for you Durable Functions Activity
```
Comment (cgillum) [inline Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs:16]:
```suggestion
            // Replace name and input with values relevant for your Durable Functions Activity
```

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+            ILogger logger = executionContext.GetLogger("DurableFunctionsOrchestrationCSharp_HttpStart");
+
+            // Function input comes from the request content.
+            string instanceId = await durableContext.Client
+                .ScheduleNewOrchestrationInstanceAsync(nameof(DurableFunctionsOrchestrationCSharp));
+
+            logger.LogInformation($"Started orchestration with ID = '{instanceId}'.");
+
+            // Returns an HTTP 202 response with an instance management payload.
+            // See https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-http-api
```
Comment (cgillum) [inline Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/DurableFunctionsOrchestrationCSharp.cs:47]:
```suggestion
            // See https://learn.microsoft.com/azure/azure-functions/durable/durable-functions-http-api#start-orchestration
```

**Review by cgillum — APPROVED** [review]
Just a couple nits but otherwise :shipit:

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/function.json (hunk 6 above)
Comment (soninaren) [reply Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/function.json:5]: You don't need this for this template. You only need this for .csx templates.

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/metadata.json [hunk 9]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (soninaren) [inline Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-Isolated/metadata.json:1]: you don't need this file either.

**Comment by jviau** [comment]
Abandoning. Will re-create with a branch from this repo so CI works.

**Pull request closed without merging**